local/hack.go

```diff
@@ -1,26 +1,71 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
 	"path/filepath"
 
-	"github.com/stackrox/rox/central/cve/datastore"
+	"github.com/stackrox/rox/central/cve/image/datastore"
+	"github.com/stackrox/rox/central/globaldb"
 	"github.com/stackrox/rox/central/globalindex"
 	"github.com/stackrox/rox/central/option"
 	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/postgres/pgconfig"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
 )
 
 func main() {
 	if features.PostgresDatastore.Enabled() {
-		return
-	}
 
-	option.CentralOptions.DBPathBase = "local/database-restore/full"
+		// PREREQUISITES TO RUNNING POSTGRES
+		// 1. Get a SQL dump of DB that has the data you want]
+		//    kubectl exec -ti -n=stackrox $(kubectl get pod -n=stackrox -l app=central-db --output 'jsonpath={.items..metadata.name}') -- sh -c '/usr/bin/pg_dumpall' > local/central_active.sql
+		// 2. Run a docker container with postgres:
+		//   docker run -it -p 5432:5432 -e POSTGRES_USER=local-postgres -e POSTGRES_PASSWORD=local-pg-password postgres:14.2-alpine
+		// 3. Restore the SQL dump from #1 to this local DB
+		//   cat local/central_active.sql |  docker exec -i $(docker ps -lq) sh -c 'PGPASSWORD=local-pg-password psql -h localhost -U local-postgres -p 5432'
```

Comment:
Can this be done with pg_restore?

Reply:
I would need to change to using pg_dump and then export as a custom format or tar in line 24. And that is giving me troubles when running via exec because the pipe is dropping lines for whatever reason. The suggestions seems to be to [write it to a local file](https://stackoverflow.com/a/51073680) and then kubectl cp it out.

Problem is that the central-db container doesn't have permissions to write. Not even to /tmp:

```
pg_dump: error: could not open TOC file "/tmo/central_active2.tar" for output: No such file or directory
```